### DISCUSSION TITLE
Tuya TS011F Plug - Max Voltage Protection?

### DISCUSSION THREAD

**Discussion by Wh1terat** [Koenkk/zigbee2mqtt#10607] [Devices]
UK voltage is supplied as 230v +10% -6% which gives a huge range of 216.2-253V.

I've bought a few TS011F plugs (tlsr8258 + bl0937) and found that they automatically turn off the relay off when mains supply >240v (which can happen pretty frequently).

At which point they cannot be turned on, mcu just wont fire the relay and more annoyingly they also won't auto recover because it's not a power loss to the plug itself.

Has anyone seen one of these within Smart Home? (I don't have a Tuya gateway) is there an option to disable this "feature" ?

**Comment by treynaer** [comment]
Hi, I have 3 plugs that randomly turn off. They are connected to the Belgium grid. I can still control them afterwards. They are also connected to Zigbee2MQTT. It's unclear to me why it happens.

**Reply by woutercoppens** [reply]
same here.

**Comment by Wh1terat** [comment]
I suspect it's the same problem @treynaer but perhaps only a momentary blip if you are able to turn them back on again immediately?

240v being only 4.25% above and although I don't know Belgium specifically I imagine it's within the threshold the grid considers "normal".

I can go days without issue then some evenings it'll fluctuate constantly and just peak over 240 causing the relay to trigger.

I've tried looking through the TuYa TLSR8258 SDK and can't see anything for over voltage, only over current from the downstream equipment.

**Comment by treynaer** [comment]
It's strainge that not all plugs go down at the same time. Grid is at 230V when a plug goes down. In total I have 8 plugs from 2 suppliers (4 per supplier). The plugs from 1 supplier are fine. The plugs from the other supplier have an issue.

I get a lot of entries like this in the logs for the 4 plugs with issues:
`debug 2022-01-28 00:51:21: No converter available for 'TS011F_plug_1' with cluster 'manuSpecificTuya_3' and type 'raw' and data '{"data":[9,87,225,2,117,236,133,41,57,0,0,0,12,0,0,0,0],"type":"Buffer"}'`

**Comment by Wh1terat** [comment]
I see ~5v difference across different parts of my house, so sometimes I only lose 1, sometimes all.

The TS011F I have has the "new" Tuya chip based on TLSR8258, looking a Tuya dev portal it shows 3-4 different chips available so perhaps different generation of plug (same part number) is different part and different code.

Some TS011F can only do power stats by reporting rather than polling (presume diff fw/hw). That looks like the reporting type ^

**Comment by Wh1terat** [comment]
I should add that mine that switch at 240v are also the ones that only support reporting.

**Comment by VeeM** [comment]
Hi all!
I have a similar problem: the plug turn off without user interaction.
In my case, it turns off only when it is under load. If no device is plugged in, it works(at least 24 hours) without turning off.
I'm not sure that the problem is relevant to the overvoltage protection. Probably it could be overcurrent protection.

Here are my traces:
```
debug 2022-02-06 20:25:10: Received Zigbee message from 'boiler_socket', type 'attributeReport', cluster 'manuSpecificTuya_3', data '{"53249":16}' from endpoint 1 with groupID 0
debug 2022-02-06 20:25:10: No converter available for 'TS011F_plug_3' with cluster 'manuSpecificTuya_3' and type 'attributeReport' and data '{"53249":16}'
debug 2022-02-06 20:25:10: Received Zigbee message from 'boiler_socket', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 1 with groupID 0
...
debug 2022-02-06 23:52:58: Received Zigbee message from 'boiler_socket', type 'attributeReport', cluster 'manuSpecificTuya_3', data '{"53249":16}' from endpoint 1 with groupID 0
debug 2022-02-06 23:52:58: No converter available for 'TS011F_plug_3' with cluster 'manuSpecificTuya_3' and type 'attributeReport' and data '{"53249":16}'
debug 2022-02-06 23:52:59: Received Zigbee message from 'boiler_socket', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 1 with groupID 0
```

The plug turns off only after "cluster 'manuSpecificTuya_3' and type 'attributeReport' and data '{"53249":16}'".
Does anybody know what "53249" attribute is?

**Reply by VeeM** [reply]
Looks like the plug should support overcurrent protection:
> Over-current alerts
> 
> (1) Use an external metering chip to measure the real-time current. If the real-time current read exceeds the set maximum operating current, an overcurrent alarm will be triggered, and the OverCurrentAlarm (private attribute) of On Off Cluster is set to 1.
> 
> (2) Turn off the relay that is being turned on, and immediately report the overcurrent alarm status OverCurrentAlarm to the gateway.
https://developer.tuya.com/en/docs/iot-device-dev/tuya-zigbee-measuring-smart-plug-access-standard?id=K9ik6zvofpzqk#title-14-Over-current%20alerts

@Wh1terat , could you please check your logs? Do you receive some attributes before the plugs turn off?

**Reply by VeeM** [reply]
I've just reproduced self turn off: plugged in two hairdryers and one iron.
And received the same logs:
```
debug 2022-02-08 23:44:18: Received Zigbee message from 'boiler_socket', type 'attributeReport', cluster 'manuSpecificTuya_3', data '{"53249":1}' from endpoint 1 with groupID 0
debug 2022-02-08 23:44:18: No converter available for 'TS011F_plug_3' with cluster 'manuSpecificTuya_3' and type 'attributeReport' and data '{"53249":1}'
debug 2022-02-08 23:44:18: Received Zigbee message from 'boiler_socket', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 1 with groupID 0
```
I have not measured the current but I'm sure that it was over 16A.

**Comment by Wh1terat** [comment]
As I said above: 
`I've tried looking through the TuYa TLSR8258 SDK and can't see anything for over voltage, only over current from the downstream equipment.`

The devices connected to my plugs aren't even pulling 1amp - my issue is not current but voltage :disappointed: 

0xD001 (53249) isn't defined in ty_iot_zigbee_sdk_tlsr8258_V1.0.2 :man_shrugging:

**Reply by VeeM** [reply]
> The devices connected to my plugs aren't even pulling 1amp - my issue is not current but voltage disappointed

Now I see, thx.

Last night my plug turned off. The load at that time was about 4A.
Probably it is the same issue as in your case.
But now plugs transmitted attribute "53249" with the value 16. 
```
debug 2022-02-09 01:31:54: Received Zigbee message from 'boiler_socket', type 'attributeReport', cluster 'manuSpecificTuya_3', data '{"53249":16}' from endpoint 1 with groupID 0
debug 2022-02-09 01:31:54: No converter available for 'TS011F_plug_3' with cluster 'manuSpecificTuya_3' and type 'attributeReport' and data '{"53249":16}'
debug 2022-02-09 01:31:54: Received Zigbee message from 'boiler_socket', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 1 with groupID 0
```

> 0xD001 (53249) isn't defined in ty_iot_zigbee_sdk_tlsr8258_V1.0.2

I hoped that this attribute could help...

**Comment by mariusg0000** [comment]
I also have two TS011F_plug_3 (_TZ3000_gjnozsaz) that disconnect the load if the main voltage exceeds 250v
![zigbeeplug](https://user-images.githubusercontent.com/4255985/160363330-7a1d635d-d093-4369-baae-6cd71b51627b.png)
.

**Reply by amayii0** [reply]
> I also have two TS011F_plug_3 (_TZ3000_gjnozsaz) that disconnect the load if the main voltage exceeds 250v ![zigbeeplug](https://user-images.githubusercontent.com/4255985/160363330-7a1d635d-d093-4369-baae-6cd71b51627b.png) .

Hi,

This is my exactly same conclusion.
Some of my plugs turn off when reaching 251V, even for a few seconds.
All are of the same model, listed as TS011F in HA.

Here are specs I got from a Belgian forum, from ORES documentation. This is used across Europe.
- Absolute minimum : 184,0 V
- Average over 10 minutes : 253,0 V
- Absolute maximum : 264,5 V

So basically the plugs are out of specs. This is particularly visible as I own solar panels and most of the neighborhood also, so these spikes are very regular, especially during summer.

I've now turned on the plugs (P10, P11, P12 in graph) and waited for sun to rise, this is pretty clear : 
![image](https://user-images.githubusercontent.com/9150547/187879052-e32f297c-be68-4361-b51b-8cc497bb81c9.png)

I've sent message to seller who simply skipped issue by replying that "yes it does turn off when over 250V", thanks Genius.
An other set of plugs from same seller do not have the issue, both 16A.
Causing issue (round, orange) : https://www.aliexpress.com/item/1005003444957152.html
Working fine (squared, white) : https://www.aliexpress.com/item/1005002344798281.html

As a workaround, I've moved these plugs for lights in living room :(
Only positive point being this defect automatically turns off lights when sun is bright around noon. Still, it's not a "feature".

I also posted at https://community.home-assistant.io/t/tuya-zigbee-plug-turning-itself-off/396484/73?u=amayii0
However this is not related to HA.

**Comment by ruant** [comment]
Just got two of these plugs in the mail yesterday.
And as soon as I connect them to some kinda load they switch off 🤦‍♂️
After some testing and moving one of them around the house, one keeps switching of without anything connected to it at all....

Other wall plugs from other vendors work fine and are reporting ~236-240v around the house at different spots.

Anything I can contribute with? Logs etc.?
Will measure the current and voltage with proper equipment while testing the plug again to see what values it cuts off on.

**Comment by kaweksl** [comment]
I also have this problem and i t seems plugs turn off outlet when over voltage is detected.

On sunny days, when photovoltaic is producing one of phases in my area oscillate around 251VAC and my UPS i toggling like crazy then. Those plugs also turn off then (on same phase). I had 3d printer connected to one off this plugs and it ruined my print :( 

Is there any way to turn that feature off ?

**Comment by ruant** [comment]
See this issue for even more people with issues with this plug
https://github.com/Koenkk/zigbee2mqtt/issues/11648

**Comment by cyrial** [comment]
Resolved here: https://github.com/Koenkk/zigbee2mqtt/discussions/13343#discussioncomment-3770375

**Comment by Wh1terat** [comment]
@cyrial Thanks for the heads up!
I can't believe this is actually resolved once and for-all ! 🎉
